crates/subspace-gateway/src/commands/http/server.rs

```diff
@@ -0,0 +1,120 @@
+use actix_web::{web, App, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Deserializer, Serialize};
+use std::default::Default;
+use std::error::Error;
+use std::sync::Arc;
+use subspace_data_retrieval::object_fetcher::ObjectFetcher;
+use subspace_data_retrieval::piece_getter::ObjectPieceGetter;
+use tracing::{debug, error, trace};
+
+pub(crate) struct ServerParameters<PG>
+where
+    PG: ObjectPieceGetter + Send + Sync + 'static,
+{
+    pub(crate) object_fetcher: ObjectFetcher<PG>,
+    pub(crate) indexer_endpoint: String,
+    pub(crate) http_endpoint: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+struct ObjectMapping {
+    #[serde(rename = "hash")]
+    hash: String,
+    #[serde(rename = "pieceIndex")]
+    piece_index: u64,
+    #[serde(rename = "pieceOffset")]
+    piece_offset: u32,
+    #[serde(rename = "blockNumber", deserialize_with = "string_to_u64")]
+    block_number: u64,
```

Comment:
Let’s import the correct type and use it?
https://github.com/autonomys/subspace/blob/aa85357117c0cefb7b9e33f7a5065ea271964843/crates/subspace-core-primitives/src/lib.rs#L138